`flake8-bugbear` (B008) flags `range()` in argument defaults, but `range` is immutable

### Summary

`B008` (`function-call-in-default-argument`) flags `range(...)` used as a default argument, even though `range` objects are immutable. Sibling immutable builtins like `slice` and `tuple` are correctly accepted, so `range` looks like an omission.

## MRE
```python
def f(x=range(3)): ...   # B008 (unexpected — range is immutable)
def g(x=slice(3)): ...   # ok
def h(x=tuple()):  ...    # ok
``` 
  
Command:

```console
$ ruff check mre.py --isolated --select B008
mre.py:1:9: B008 Do not perform function call `range` in argument defaults; ...
Found 1 error.
```
  
**Expected:** no diagnostic for `range()`, consistent with `slice()`/`tuple()`.
**Actual:** `range()` is flagged.

Playground link: https://play.ruff.rs/e8b9dd7a-0646-4158-9a87-3adef4d097e6

## Likely cause
  
`range` is missing from `is_immutable_return_type` (`crates/ruff_python_stdlib/src/typing.rs`), which lists `slice`, `tuple`, `frozenset` etc. Notably, `range` *is* already treated as immutable in `is_immutable_non_generic_type` in the same file, so the two are inconsistent.
  
This mirrors #21565 / #21823 ("Accept immutable slice default arguments"); `range` appears to be the same class of fix.

### Version

ruff 0.16.0

--- a/crates/ruff_linter/resources/test/fixtures/flake8_bugbear/B006_B008.py
+++ b/crates/ruff_linter/resources/test/fixtures/flake8_bugbear/B006_B008.py
@@ -98,7 +98,7 @@ def dont_forget_me(value=collections.deque()):
     ...
 
 
-# N.B. we're also flagging the function call in the comprehension
+# B006 still flags mutable comprehension defaults.
 def list_comprehension_also_not_okay(default=[i**2 for i in range(3)]):
     pass
 

--- a/crates/ruff_python_stdlib/src/typing.rs
+++ b/crates/ruff_python_stdlib/src/typing.rs
@@ -324,6 +324,7 @@ pub fn is_immutable_return_type(qualified_name: &[&str]) -> bool {
                     | "float"
                     | "frozenset"
                     | "int"
+                    | "range"
                     | "str"
                     | "tuple"
                     | "slice"

